### ISSUE TITLE
Calendar EventSource using cron schedule seems to trigger twice

### ISSUE THREAD

**Issue by juliev0** [argoproj/argo-events#1828]
**Describe the bug**
At least some of the time, if not every time, in the log I see "dispatching calendar event" and "succeeded to publish an event" happening twice at approximately the same time.

**To Reproduce**
I'm using this yaml:
```
apiVersion: argoproj.io/v1alpha1
kind: EventSource
metadata:
  name: calendar
spec:

  #eventBusName: jetstream-ex
  replicas: 2
  calendar:
    example:
      schedule: "* * * * *"
    example2:
      schedule: "08 23 * * *"
```

**Expected behavior**
calendar.example should only trigger once a minute

**Environment (please complete the following information):**
Latest master: commit ed5ff514855c10de11541bca41b1e202668f662c
k3d on mac

Logs:

{"level":"info","ts":1649463600.0004694,"logger":"argo-events.eventsource","caller":"calendar/start.go:209","msg":"dispatching calendar event...","eventSourceName":"calendar","eventSourceType":"calendar","eventName":"example"}
{"level":"info","ts":1649463600.0065558,"logger":"argo-events.eventsource","caller":"eventsources/eventing.go:537","msg":"succeeded to publish an event","eventSourceName":"calendar","eventName":"example","eventSourceType":"calendar","eventID":"65663161373566362d663037342d343631632d393631312d353435363131313035333162"}
**{"level":"info","ts":1649463600.0067806,"logger":"argo-events.eventsource","caller":"calendar/start.go:249","msg":"expected next calendar event","eventSourceName":"calendar","eventSourceType":"calendar","eventName":"example","time":"2022-04-09 00:21:00 +0000 UTC"}**
{"level":"info","ts":1649463659.9333835,"logger":"argo-events.eventsource","caller":"calendar/start.go:209","msg":"dispatching calendar event...","eventSourceName":"calendar","eventSourceType":"calendar","eventName":"example"}
**{"level":"info","ts":1649463659.937736,"logger":"argo-events.eventsource","caller":"eventsources/eventing.go:537","msg":"succeeded to publish an event","eventSourceName":"calendar","eventName":"example","eventSourceType":"calendar","eventID":"64633237643831342d333361622d343430362d383838352d373634386337333735336138"}**
**{"level":"info","ts":1649463659.9379187,"logger":"argo-events.eventsource","caller":"calendar/start.go:249","msg":"expected next calendar event","eventSourceName":"calendar","eventSourceType":"calendar","eventName":"example","time":"2022-04-09 00:21:00 +0000 UTC"}**
{"level":"info","ts":1649463660.0009928,"logger":"argo-events.eventsource","caller":"calendar/start.go:209","msg":"dispatching calendar event...","eventSourceName":"calendar","eventSourceType":"calendar","eventName":"example"}
**{"level":"info","ts":1649463660.0043728,"logger":"argo-events.eventsource","caller":"eventsources/eventing.go:537","msg":"succeeded to publish an event","eventSourceName":"calendar","eventName":"example","eventSourceType":"calendar","eventID":"64356339633166662d333539392d346237312d626364302d343133353766353063626565"}**

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by juliev0** [comment]
Tried code from 1/26 and it behaves this way too. Even if I use a specific minute rather than every minute.

**Comment by juliev0** [comment]
@whynowy I was not able to reproduce this in a simple example unfortunately. What do you think about merging the workaround?

**Comment by whynowy** [comment]
@juliev0 - what do you mean by `simple example`?

**Comment by juliev0** [comment]
@whynowy  Actually, it wasn't that simple. I started with a simple timer to see if it would fire early. It didn't, so I created a bunch of goroutines to do it at the same time. Still no timers fired early. So, I had each goroutine operating in a loop. No timer ever fired early. I also created a Docker image and imported it into k3d and tried it there (attaching code).
[doubletrigger.zip](https://github.com/argoproj/argo-events/files/8476330/doubletrigger.zip)
